pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -92,6 +96,30 @@ type UpdateStrategy struct {
 	*RegistryPoll `json:"registryPoll,omitempty"`
 }
 
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface.
+// UnmarshalJSON checks to see the provided polling interval is valid. If so, use it, if not set it to the default interval.
+func (u *UpdateStrategy) UnmarshalJSON(data []byte) error {
+	var updateStrategy map[string]map[string]string
+	if err := json.Unmarshal(data, &updateStrategy); err != nil {
+		return err
+	}
+	if updateStrategy == nil {
+		return nil
+	}
+
+	registryPoll := &RegistryPoll{}
+	if rp, ok := updateStrategy["registryPoll"]; ok {
+		duration, err := time.ParseDuration(rp["interval"])
+		if err != nil {
+			registryPoll.Interval = &metav1.Duration{Duration: defaultPollingDuration}
```

Comment:
I think passing that error along to the status is a good option, but I'm not sure how we can do that inside of this unmarshaling function. We do not have access to the status inside the custom unmarshaler. Is there an easy way to set the status in this context? I feel like the pattern is controllers should set the status, this implementation is almost like a mutating webhook that changes a parameter to a default.

It may surprise users but this is an edge-case to begin with. I think we can document it as part of the existing polling docs.

Reply:
The other option is to simply strip out the invalid interval and set updateStrategy to nil, which is also possible. If that seems less surprising we could do that instead.

Reply:
Yeah, that's a point - I'm not entirely sure how we'd bubble this back up except returning a custom error that we check at the call site (or whether that's even possible). I'm not sure I have a super strong opinion on either approach: documentation sounds reasonable, but you'd still run into the UX gap when debugging further, vs. let's try and follow the last surprising approach (i.e. https://en.wikipedia.org/wiki/Principle_of_least_astonishment). Might be worth trying to get some other opinions?